include/concurrencpp/threads/cache_line.h

```diff
@@ -6,9 +6,16 @@
 #include <new>
 
 #if !defined(CRCPP_MAC_OS) && defined(__cpp_lib_hardware_interference_size)
-#    define CRCPP_CACHE_LINE_ALIGNMENT std::hardware_destructive_interference_size
+#    if defined(CRCPP_GCC_COMPILER) 
+#        pragma GCC diagnostic push
+#        pragma GCC diagnostic ignored "-Winterference-size"
+#    endif
+constexpr inline std::size_t CRCPP_CACHE_LINE_ALIGNMENT = std::hardware_destructive_interference_size;
+#    if defined(CRCPP_GCC_COMPILER) 
+#        pragma GCC diagnostic pop
+#    endif
 #else
-#    define CRCPP_CACHE_LINE_ALIGNMENT 64
+constexpr inline std::size_t CRCPP_CACHE_LINE_ALIGNMENT = 64;
 #endif

```

Comment:
I turned this preprocessor definition into a `constexpr inline` in order to be able to wrap it in `#pragma GCC diagnostic push/pop`.